### DISCUSSION TITLE
Surround multi-line

### DISCUSSION THREAD

**Discussion by 9mm** [nvim-mini/mini.nvim#922] [Q&A]
Im looking at `surround.mini`.

Imagine this text:

```
awoefawef
waefawefawef
awefawefij foaweif aweofi
waeifjoaw
[a]wef
```
cursor is on `[a]`. Press `<C-v>gg$sa"` and I get this:

```
"awoefawef
"waefawefaw"ef
"awefawefij" foaweif aweofi
"waeifjoaw
"awef
```

Expected is:

```
"awoefawef"
"waefawefawef"
"awefawefij foaweif aweofi"
"waeifjoaw"
"awef"
```

I notice this is possible on all the vim based plugins (tpope etc) but not on the two nvim based plugins. Is there a way to do this?

**Comment by echasnovski** [comment]
This is similar to the use case discussed in #770.

This particular case of surrounding blockwise selection together with `$` will not be supported for at least the following reasons:
- Actual implementation is not trivial while 'mini.surround' is already quite a big and complicated module.
- It can be replicated with a macro. In this particular case: record macro with `qq`, `0sa$"`, `k`, `q`; replay with `4Q`.

**Reply by 9mm** [reply]
Hmm ok. Is the reason its supported on all the vimcript plugins but all the neovim plugins say its impossible/complicated because the neovim interface makes it a challenge while vimscript it is not? or do you think its tricky in both

**Reply by echasnovski** [reply]
No, not really. All basic Vimscript tools (at least before Vim9script) are available in Neovim.

It is the fact that this specific example needs to be treated as special case with possibly not great workarounds (here is what looks like a relevant part of [vim-sandwich](https://github.com/machakann/vim-sandwich/blob/74cf93d58ccc567d8e2310a69860f1b93af19403/autoload/operator/sandwich.vim#L83-L107); I don't quite understand 'vim-surround' code) while being quite easily replaced with macro.
